app/celery/process_delivery_status_result_tasks.py

```diff
@@ -1,13 +1,296 @@
+    current_app.logger.info(
+        "Processing Notification Delivery Status. | reference=%s | notification_status=%s | "
+        "number_of_message_parts=%s | price_in_millicents_usd=%s",
+        reference,
+        notification_status,
+        number_of_message_parts,
+        price_in_millicents_usd,
+    )
+    return payload, reference, notification_status, number_of_message_parts, price_in_millicents_usd
+
+
+def _calculate_pricing(price_in_millicents_usd, notification, notification_status, number_of_message_parts):
+    """ Calculate pricing """
+    if price_in_millicents_usd > 0.0:
+        notification.status = notification_status
+        notification.segments_count = number_of_message_parts
+        notification.cost_in_millicents = price_in_millicents_usd
+        dao_update_notification(notification)
+    else:
+        # notification_id -  is the UID in the database for the notification
+        # status - is the notification platform status generated earlier
+        update_notification_status_by_id(
+            notification_id=notification.id, status=notification_status
+        )
+
+
+def _get_notification_platform_status(self, provider, body, sqs_message):
+    """ Performs a translation on the body """
+    notification_platform_status = None
+    try:
+        notification_platform_status = provider.translate_delivery_status(body)
+    except (ValueError, KeyError) as e:
+        current_app.logger.error("The event stream body could not be translated.")
+        current_app.logger.exception(e)
+        current_app.logger.debug(sqs_message)
+        self.retry(queue=QueueNames.RETRY)
+
+    current_app.logger.info(
+        "retrieved delivery status: %s", notification_platform_status
+    )
+
+    # notification_platform_status cannot be None
+    if notification_platform_status is None:
+        current_app.logger.error("Notification Platform Status cannot be None")
+        current_app.logger.debug(body)
+        self.retry(queue=QueueNames.RETRY)
+
+    return notification_platform_status
+
+
+def _get_include_payload_status(self, notification):
+    """ Determines where payload should be included in delivery status callback data"""
+    include_payload_status = False
+
+    try:
+        include_payload_status = dao_get_callback_include_payload_status(
+            notification.service_id,
+            notification.notification_type
+        )
+
+    except NoResultFound:
+        # It is acceptable for this exception to happen but we want to log it
+        current_app.logger.info("ServiceCallback include_payload for notification is unavailable or false")
+        current_app.logger.debug(notification)
+
+    except (AttributeError, TypeError) as e:
+        current_app.logger.error("Could not determine include_payload property for ServiceCallback.")
+        current_app.logger.exception(e)
+        current_app.logger.debug(notification)
+        self.retry(queue=QueueNames.RETRY)
```

Comment:
Please revise this to not rely on exceptions for an entry not in the DB. This will throw an inordinate amount of exceptions with normal use.